leetcode_export/__main__.py

```diff
@@ -153,6 +166,16 @@ def main():
         if args.language and submission.lang not in args.language:
             continue
 
+        if submission.title_slug in title_slug_to_extension and \
+            submission.extension in title_slug_to_extension[submission.title_slug] and \
+                args.only_latest_submission:
+            logging.info(f"Saving only the latest submission of the question {submission.title_slug} with "
+                         f"{submission.extension} extension, skipping it")
+            continue
+        else:
+            title_slug_to_extension[submission.title_slug] = set()
```

Comment:
This is good but it's missing an edge case, refactoring the code a bit will fix it.

Consider the following sequence of submission programming languages: cpp, java, cpp, cpp.

This code will download "cpp, java, cpp" because in the second iteration (java) the condition will resolve to false and `title_slug_to_extension[submission.title_slug]` will be substituted with a new empty set

Reply:
Also, I'm not sure if we should rely on the API returning the submissions in reverse chronological order. As it's LeetCode private API we don't have any guarantee on it, they could change its behaviour anytime

Reply:
> This is good but it's missing an edge case, refactoring the code a bit will fix it.
> 
> Consider the following sequence of submission programming languages: cpp, java, cpp, cpp.
> 
> This code will download "cpp, java, cpp" because in the second iteration (java) the condition will resolve to false and `title_slug_to_extension[submission.title_slug]` will be substituted with a new empty set

Thanks for pointing out. I have fixed it and will push it soon.

Reply:
> Also, I'm not sure if we should rely on the API returning the submissions in reverse chronological order. As it's LeetCode private API we don't have any guarantee on it, they could change its behaviour anytime

It could be a problem. In the newly committed code, I got all submissions and sorted them by the `date_formatted`.

Reply:
Thank you for the follow up!

After further consideration, I think it's best to write the submission as soon as possible otherwise users might think that the script doesn't work if it takes 5-10 minutes before a submission appears in the output folder.

> Note: the script might take several minutes to execute because of LeetCode API rate limits

Sorry for the confusion, I made the required changes and added a warning message prompting users to create a GitHub issue if the script detects that submissions are not in reverse chronological order.